src/analysis/normed_space/star/multiplier.lean

```diff
@@ -0,0 +1,429 @@
+
+instance : has_smul S 𝓜(𝕜, A) :=
+{ smul := λ s a,
+  { to_prod := s • a.to_prod,
+    central := λ x y, show (s • a.snd) x * y = x * (s • a.fst) y,
+      by simp only [continuous_linear_map.smul_apply, mul_smul_comm, smul_mul_assoc, central] } }
+
+@[simp] lemma smul_fst (k : 𝕜) (a : 𝓜(𝕜, A)) : (k • a).fst = k • a.fst := rfl
+@[simp] lemma smul_snd (k : 𝕜) (a : 𝓜(𝕜, A)) : (k • a).snd = k • a.snd := rfl
+
+end scalars
+
+instance : has_one 𝓜(𝕜, A) := ⟨⟨1, λ x y, rfl⟩⟩
+
+instance : has_mul 𝓜(𝕜, A) :=
+{ mul := λ a b,
+  { to_prod := (a.fst.comp b.fst, b.snd.comp a.snd),
+    central := λ x y, show b.snd (a.snd x) * y = x * (a.fst (b.fst y)),
+      by simp only [central] } }
+
+instance : has_nat_cast 𝓜(𝕜, A) :=
+  { nat_cast := λ n, ⟨n, λ x y, by simp only [←nat.smul_one_eq_coe, prod.smul_fst, prod.smul_snd,
+      prod.fst_one, prod.snd_one, smul_apply n 1, one_apply, mul_smul_comm, smul_mul_assoc]⟩ }
+
+instance : has_int_cast 𝓜(𝕜, A) :=
+  { int_cast := λ n, ⟨n, λ x y, by simp only [←int.smul_one_eq_coe, prod.smul_fst, prod.smul_snd,
+      prod.fst_one, prod.snd_one, smul_apply n 1, one_apply, mul_smul_comm, smul_mul_assoc]⟩ }
+
+instance : has_pow 𝓜(𝕜, A) ℕ :=
+  { pow := λ a n, ⟨a.to_prod ^ n, λ x y,
+    begin
+      induction n with k hk generalizing x y,
+      refl,
+      rw [prod.pow_snd, prod.pow_fst] at hk ⊢,
+      rw [pow_succ a.snd, mul_apply, a.central, hk, pow_succ' a.fst, mul_apply],
+    end⟩, }
+
+@[simp] lemma add_fst (a b : 𝓜(𝕜, A)) : (a + b).fst = a.fst + b.fst := rfl
+@[simp] lemma add_snd (a b : 𝓜(𝕜, A)) : (a + b).snd = a.snd + b.snd := rfl
+@[simp] lemma zero_fst : (0 : 𝓜(𝕜, A)).fst = 0 := rfl
+@[simp] lemma zero_snd : (0 : 𝓜(𝕜, A)).snd = 0 := rfl
+@[simp] lemma neg_fst (a : 𝓜(𝕜, A)) : (-a).fst = -a.fst := rfl
+@[simp] lemma neg_snd (a : 𝓜(𝕜, A)) : (-a).snd = -a.snd := rfl
+@[simp] lemma sub_fst (a b : 𝓜(𝕜, A)) : (a - b).fst = a.fst - b.fst := rfl
+@[simp] lemma sub_snd (a b : 𝓜(𝕜, A)) : (a - b).snd = a.snd - b.snd := rfl
+@[simp] lemma one_fst : (1 : 𝓜(𝕜, A)).fst = 1 := rfl
+@[simp] lemma one_snd : (1 : 𝓜(𝕜, A)).snd = 1 := rfl
+@[simp] lemma mul_fst (a b : 𝓜(𝕜, A)) : (a * b).fst = a.fst * b.fst := rfl
+@[simp] lemma mul_snd (a b : 𝓜(𝕜, A)) : (a * b).snd = b.snd * a.snd := rfl
+@[simp] lemma nat_cast_fst (n : ℕ) : (n : 𝓜(𝕜 , A)).fst = n := rfl
+@[simp] lemma nat_cast_snd (n : ℕ) : (n : 𝓜(𝕜 , A)).snd = n := rfl
+@[simp] lemma int_cast_fst (n : ℤ) : (n : 𝓜(𝕜 , A)).fst = n := rfl
+@[simp] lemma int_cast_snd (n : ℤ) : (n : 𝓜(𝕜 , A)).snd = n := rfl
+@[simp] lemma pow_fst (n : ℕ) (a : 𝓜(𝕜, A)) : (a ^ n).fst = a.fst ^ n := rfl
+@[simp] lemma pow_snd (n : ℕ) (a : 𝓜(𝕜, A)) : (a ^ n).snd = a.snd ^ n := rfl
+
+/-- The natural injection from `double_centralizer.to_prod` except the second coordinate inherits
+`mul_opposite.op`. The ring structure on `𝓜(𝕜, A)` is the pullback under this map. -/
+def to_prod_mop : 𝓜(𝕜, A) → ((A →L[𝕜] A) × (A →L[𝕜] A)ᵐᵒᵖ) :=
```

Comment:
I would avoid `mop` in the name. This is the name of the notation, not of the type. But I agree `mul_opposite` is a long name, so maybe something like this?
```suggestion
def to_End : 𝓜(𝕜, A) → module.End 𝕜 A × (module.End 𝕜 A)ᵐᵒᵖ :=
```
with `module.End` replaced with the `continuous_linear_map` analog (I assume it exists somewhere?)